File: 00-每日一题/20200404_345.cpp
//
// Created by zanbo on 2020/4/4.
//

class Solution {
public:
    string reverseVowels(string s) {
        int n=s.size();
        if(n<=1) return s;
        int i=0;
        int j=n-1;
        while(i<j)
        {
            while(i<j && tolower(s[i])!='a' && tolower(s[i])!='e' && tolower(s[i])!='i' && tolower(s[i])!='o' &&tolower(s[i])!='u')
            {
                i++;
            }
            while(i<j && tolower(s[j])!='a' && tolower(s[j])!='e' && tolower(s[j])!='i' && tolower(s[j])!='o' &&tolower(s[j])!='u')
            {
                j--;
            }
            if(i<j)
            {
                char temp = s[i];
                s[i] = s[j];
                s[j] = temp;
                i++;
                j--;
            }
        }
        return s;
    }
};

File: 00-每日一题/20200404_345.py
# 对撞指针
class Solution(object):
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        vowels = set(['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'])
        s = list(s)
        length = len(s)
        left = 0
        right = length - 1
        while left < right:
            if s[left] not in vowels:
                left += 1
                continue
            if s[right] not in vowels:
                right -= 1
                continue

            s[left], s[right] = s[right], s[left]
            left += 1
            right -= 1
        return ''.join(s)


if __name__ == '__main__':

    obj = Solution()
    while True:
        s = input().strip()
        res = obj.reverseVowels(s)
        print(res)